libs/platform/src/lib/list/base-list-item.ts

```diff
@@ -143,152 +146,177 @@ export class BaseListItem extends BaseComponent implements OnInit, AfterViewChec
+     */
+    rowSelection: boolean;
+
     /**
      * @hidden
-     * Whether listitem is selected binded to template */
+     * Whether listitem is selected binded to template
+     */
     _selected: boolean;
+
     /**
      * @hidden
      * get the focused element for key manager
      */
     _focused: boolean;
 
+    /**
+     * @hidden
+     * radio button selected value binded to template
+     */
+    private _selectionValue: string;
+
     /** @hidden */
     constructor(
         protected _changeDetectorRef: ChangeDetectorRef,
-        public itemEl: ElementRef,
-        protected _listConfig: ListConfig,
-        @Optional() private router: Router
+        public itemEl: ElementRef<HTMLElement>,
+        protected _listConfig: ListConfig
     ) {
         super(_changeDetectorRef);
     }
 
-    /** setter and getter for _link */
-    @Input('link')
-    get routerLink(): string {
-        return this.link;
-    }
-
-    set routerLink(value: string) {
-        this.link = value;
-    }
-
-    /** @hidden
-     * radio button selected value binded to template */
-    _selectionValue: string;
-
-    @Input('selectionValue')
-    get selectionValue(): string {
-        return this._selectionValue;
-    }
-
-    set selectionValue(value: string) {
-        this._selected = false;
-        this._selectionValue = value;
-        if (this._selectionValue !== undefined && this.selectionValue !== null) {
-            this._selected = this.value === this._selectionValue;
-        }
-    }
-
-    /** @hidden */
-    /** Show navigation for single list*/
+    /**
+     * @hidden
+     * Show navigation for single list
+     */
     ngOnInit(): void {
         this.id = `fdp-list-item-${nextListItemId++}`;
     }
 
-    /** @hidden */
-    /** To detect changes from parent-listbox to list item
+    /**
+     * @hidden
+     * To detect changes from parent-listbox to list item
      * for example single, multi option selection those details
      * will be deducted in list item
      */
     ngAfterViewChecked(): void {
-        this._changeDetectorRef.detectChanges();
-        const currentitem = this.itemEl.nativeElement.querySelector('li');
-        if (currentitem) {
-            currentitem.setAttribute('role', 'option');
-            if (currentitem.parentNode) {
-                currentitem.parentNode.removeAttribute('title');
-                currentitem.parentNode.removeAttribute('aria-label');
-                if (this.selectRow || this.selectionMode === 'multi' || this.selectionMode === 'single') {
-                    currentitem.parentNode.setAttribute('aria-selected', this._selected ? this._selected : false);
-                }
-            }
+        const currentitem: HTMLLIElement = this.itemEl.nativeElement.querySelector('li');
+        if (!currentitem) {
+            return;
+        }
+
+        currentitem.setAttribute('role', 'option');
+        const parentNode = currentitem.parentNode as HTMLElement;
```

Comment:
I would suggest slightly safer approach
```suggestion
        const parentNode = currentitem.parentNode instanceof HTMLElement && currentitem.parentNode;
```